src/parse.rs

```diff
@@ -0,0 +1,319 @@
+use crate::config::{ParserSettings, LINK_REGEX};
+use crate::document::{
+    code::{CodeBlock, Line, Source},
+    text::TextBlock,
+    transclusion::Transclusion,
+    Document, Node,
+};
+use crate::util::{Fallible, TryCollectExt};
+use std::error::Error;
+use std::fmt::Write;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+#[allow(clippy::nonminimal_bool)]
+pub fn parse(input: &str, path: &Path, settings: &ParserSettings) -> Fallible<Document> {
+    #[derive(Default)]
+    struct State {
```

Comment:
I don' think this type really adds anything. Why not go for `let mut state = None;` instead?

Reply:
See #97.